spi/src/main/java/org/mapstruct/spi/AccessorNamingStrategy.java

```diff
@@ -0,0 +1,134 @@
+/**
+ *  Copyright 2012-2015 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.spi;
+
+/**
+ * A service provider interface for the mapping between method names and properties.
+ *
+ * @author Christian Schuster
+ */
+public interface AccessorNamingStrategy {
+
+    /**
+     * Set the default {@link AccessorNamingStrategy} implementation. Custom implementations may use it to
+     * keep parts of the default behavior. This method is invoked before any other method of the interface.
+     * <p>
+     * Default implementation: Does nothing.
+     *
+     * @param defaultAccessorNamingStrategy The default implementation.
+     */
+    void setDefaultAccessorNamingStrategy(AccessorNamingStrategy defaultAccessorNamingStrategy);
+
+    /**
+     * Determine if a method name defines a getter with a non-boolean return type.
+     * <p>
+     * Default implementation: Method name starts with "get" and is longer than 3 characters.
+     *
+     * @param methodName The method name.
+     * @return {@code true} if the method name can be a non-boolean getter, {@code false} otherwise.
+     */
+    boolean isNonBooleanGetterName(String methodName);
```

Comment:
The `Executables` class does the method signature checks, and those are different for boolean and non-boolean getters. We could pass the whole `ExecutableElement` and do the signature check in the SPI.